packages/transition-frontend/src/components/forms/node/TransitNodeEdit.tsx

```diff
@@ -142,6 +143,43 @@ class TransitNodeEdit extends SaveableObjectForm<Node, NodeFormProps, NodeFormSt
             transitNodes1000mRadius: turfFeatureCollection(geojson.geometry ? [geojson] : []),
             transitNodesRoutingRadius: turfFeatureCollection(geojson.geometry ? [geojson] : [])
         });
+        serviceLocator.socketEventManager.emit(
+            'osm.streetsAroundPoint',
+            this.state.object.toGeojson(),
+            100,
+            (status: Status.Status<GeoJSON.Feature[]>) => {
+                if (Status.isStatusOk(status)) {
+                    const streetsGeojsonFeatures: GeoJSON.Feature[] = status.result;
+                    if (streetsGeojsonFeatures.length < 10) {
+                        // limit streets count to 10, because we will find all possible intersections
+                        const uniqStreetNames: string[] = _uniq(
+                            streetsGeojsonFeatures
+                                .filter((f) => f.properties?.name !== undefined)
+                                .map((f) => f.properties?.name)
+                        );
+                        const possibleIntersectionNames: string[] = [];
+                        for (let i = 0, countI = uniqStreetNames.length; i < countI; i++) {
+                            for (let j = i + 1, countJ = uniqStreetNames.length; j < countJ; j++) {
+                                const streetName1 = uniqStreetNames[i];
+                                const streetName2 = uniqStreetNames[j];
+                                if (streetName1 !== streetName2) {
+                                    possibleIntersectionNames.push(`${streetName1} / ${streetName2}`);
```

Comment:
Should you push the 2 possible permutations? like Villeray/St-Michel, St-Michel/Villeray?

Reply:
done